File: app/src/main/java/com/example/user/biblejournal/view/MyClickableSpan.java
package com.example.user.biblejournal.view;

import android.text.style.ClickableSpan;
import android.view.View;

import androidx.annotation.NonNull;

public class MyClickableSpan extends ClickableSpan {
    private ClickableSpanListener clickListener;

    public MyClickableSpan(ClickableSpanListener clickListener) {
        this.clickListener = clickListener;
    }

    public interface ClickableSpanListener {
        void onClickableSpanClick();
    }

    @Override
    public void onClick(@NonNull View widget) {
        clickListener.onClickableSpanClick();
    }

}

File: app/src/main/java/com/example/user/biblejournal/writer/view/MyClickableSpan.kt
package com.example.user.biblejournal.writer.view

import android.text.TextPaint
import android.text.style.ClickableSpan
import android.view.View
import com.example.user.biblejournal.reader.repository.VerseAddress

class MyClickableSpan(private val clickListener: ClickableSpanListener, private val verseAddress: VerseAddress) : ClickableSpan() {

    interface ClickableSpanListener {
        fun onClickableSpanClick(verseAddress: VerseAddress)
    }

    override fun onClick(widget: View) {
        clickListener.onClickableSpanClick(verseAddress)
    }

    override fun updateDrawState(ds: TextPaint) {
        super.updateDrawState(ds)
        ds.isUnderlineText = false
    }

}
